### PR TITLE
authentication

### PR DESCRIPTION
**Pull request by znjs**
Added authentication functionality

### CODE REVIEW

**Review by Launchpad5682 — APPROVED** [review]
* Good folder structure.
* Clean code.
Good to merge.

File: src/context/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        let res = await axios.post("/api/auth/login", {
+          email: email,
+          password: password,
+        });
+        if (res.status === 200) {
+          const {
+            data: { encodedToken, foundUser },
+          } = res;
+          userDispatch({
+            type: "UPDATE_TOKEN_AND_DATA",
```
Comment (ScoobyDubeyDo) [inline src/context/auth-context.js]: **suggestion**: you can you constant values in place of strings, for bug-free experience

File: src/index.js [hunk 2]
```diff
@@ -4,14 +4,21 @@ import "./index.css";
...
   <React.StrictMode>
     <BrowserRouter>
-      <App />
+      <AuthProvider>
+        <ProductProvider>
+          <FilterProvider>
+            <App />
+          </FilterProvider>
+        </ProductProvider>
+      </AuthProvider>
```
Comment (ScoobyDubeyDo) [inline src/index.js:21]: **suggestion**: I would recommend you to make a final provider which will be the only wrapper of the app.

Comment (ScoobyDubeyDo) [inline src/index.js:21]: **suggestion**: I would recommend you to make a final provider which will be the only wrapper of the app.

**Review by ScoobyDubeyDo — APPROVED** [review]
Everything looks good, I have given a few suggestions, try to fix them before your next PR

**Pull request merged**